physical/consul/consul.go

```diff
@@ -224,6 +224,19 @@ func NewConsulBackend(conf map[string]string, logger log.Logger) (physical.Backe
 		if logger.IsDebug() {
 			logger.Debug("config address set", "address", addr)
 		}
+
+		// Copied from the Consul API module; set the Scheme based on
+		// the protocol field if address looks ike a URL.
+		// This can enable the TLS configuration below.
+		parts := strings.SplitN(addr, "://", 2)
```

Comment:
Probably better to use url.Parse, then use the scheme from that.

Reply:
I thought about that but I got scared off by

"Trying to parse a hostname and path without a scheme is invalid but may not necessarily return an error, due to parsing ambiguities."

I ran a quick test and `myhostname:8080` parses as

```
&url.URL{
    Scheme:     "myhostname",
    Opaque:     "8080",
    User:       (*url.Userinfo)(nil),
    Host:       "",
    Path:       "",
    RawPath:    "",
    ForceQuery: false,
    RawQuery:   "",
    Fragment:   "",
}
```

So we could use url.Parse but then we'd have to double-check the result in the common case of people who had actually followed the documentation correctly.

Reply:
Hm, GJ url.Parse. Oh well. Looks fine then.